### DISCUSSION TITLE
How to debug generate process when showing error 404

### DISCUSSION THREAD

**Discussion by TeddyHuang-00** [nuxt/content#2233] [Q&A]
`nuxt generate` with `ssr: false` works just fine, but when I set `ssr: true` and the prerender step will complain that some payload (I guess?) json files are missing. And since their names are sort of hash-ed (e.g. `/api/_content/query/wxmlyJ2dlX.1692374233767.json`), how can I debug the generate process to see who and where is making such queries so that I can fix things up?

It is not a huge problem as I've got `ssr: false` as a workaround, but giving up SSR seems to be a huge loss for a document driven site.

**Comment by TeddyHuang-00** [comment]
Yeah, never mind the original question on how to debug command, I found that this can be done quite easily with VS Code debug session. Now the problem is the the error is generated from
```mjs
await nitro.hooks.callHook("prerender:routes", routes);
```
And I need to look for where the hook is registered and what it does....

**Comment by TeddyHuang-00** [comment]
I found that the problem lies in the header URLs under the name `x-nitro-prerender`. And three of my routes generated one faulty query JSON each, and all of them happened to be only include a div tag in the template. Still testing out solutions....

---

EDIT: BTW, this is generated at the line 98-103 in the subfunction `generateRoute` of function `prerender` in file `nitropack@[ver]/node_modules/nitropack/dist/shared/nitro.[xxx].mjs`.

**Reply by NJordan72** [reply]
Did you ever solve this?  I'm having the same problem, interestingly with the same hash `wxmlyJ2dlX`.

**Reply by TeddyHuang-00** [reply]
Well I did solve the issue, but I have done that through many trials and errors, so that I cannot recall what steps are the solution to the problem. But I did remember upgrading to a newer version of Nuxt and it produces the exact error and I believe that is related to some pages being needed without definition in the pages directory. I strongly suggest you check your Nuxt version and upgrade if you can, and also see if you are referring to `/subdir/` while using file structure like `pages/subdir/[slug].vue` without defining `pages/subdir/index.vue`. Hope this is helpful to you!
